testsuite/tests/parallel/type_injectivity.ml

```diff
@@ -0,0 +1,8 @@
+(* TEST
+* hasunix
+include unix
+** bytecode
+** native
+*)
```

Comment:
Notes on this test:
- I don't think it should be in `parallel/`, which is a directory with way too much stuff, related to the parallel runtime; would you create a new directory `tests/lib-domain` for the test?
- could you simplify the header? I'm not sure that you need `hasunix` (I suppose you reused it from some other header?), and I think there are shorter ways to test with both bytecode and native -- this should be the default. In fact just `(* TEST *)` may work.